src/cli/bin.ts

```diff
@@ -1,8 +1,71 @@
 import yargs from "yargs";
 import _capitalize from "lodash/capitalize";
+import _cloneDeep from "lodash/cloneDeep";
 import fs from "fs-extra";
 import { migrate_43_to_50 } from "../4.3_to_5.0/migrate_43_to_50";
 import path from "path";
+import { ContentRecord, DataModel } from "@activeviam/activeui-sdk-5.0";
+import { ErrorReport, OutcomeCounters } from "../4.3_to_5.0/migration.types";
+import _fromPairs from "lodash/fromPairs";
+import { emptyUIFolder } from "@activeviam/content-server-initialization-5.0";
+
+type MigrationFunction = (
+  legacyUIFolder: ContentRecord,
+  {
+    migratedUIFolder,
+    counters,
+    errorReport,
+    servers,
+    keysOfWidgetPluginsToRemove,
+    legacyPivotFolder,
+    doesReportIncludeStacks,
+  }: {
+    migratedUIFolder: ContentRecord;
+    counters: OutcomeCounters;
+    errorReport: ErrorReport;
+    servers: {
+      [serverKey: string]: {
+        dataModel: DataModel;
+        url: string;
+      };
+    };
+    keysOfWidgetPluginsToRemove?: string[] | undefined;
+    doesReportIncludeStacks: boolean;
+    legacyPivotFolder?: ContentRecord<any> | undefined;
+  },
+) => Promise<void>;
+
+const migrationFunctions: [
+  versionFrom: string,
+  migrationFunction?: MigrationFunction,
+][] = [["4.3", migrate_43_to_50], ["5.0"]];
+const versions = migrationFunctions.map(([version]) => version);
+const fromVersions = versions.slice(0, -1);
+const toVersions = versions.slice(1);
+
+const compose =
```

Comment:
I think the recursive implementation adds unnecessary complexity here.
I would go for `reduce` or a `for` loop, and actually we could inline it instead of going for a `compose` abstraction since it'll be used in only one place.